### DISCUSSION TITLE
Width of buttons/containers changes based on largest button

### DISCUSSION THREAD

**Discussion by sina--** [elkowar/eww#853] [Q&A] [answered]
Hey everyone,

I'm trying to configure a a bar to use with Hyprland, and got things mostly setup how I'd like, however there are a couple quirks I am unable to figure out, that I was hoping I could get some help with. For my workspace indicator I'm using hyprland-workspaces, and basically just copied the widget from their config.

```
(deflisten workspace0 "hyprland-workspaces `hyprctl monitors -j | jq -r \".[0].name\"`")

(defwidget workspaces0 []
  (eventbox :onscroll "hyprctl dispatch workspace `echo {} | sed 's/up/+/\' | sed 's/down/-/'`1"
    (box :class "windowManager" :halign "start" :spacing 10 :space-evenly: "false"
      (for i in workspace0
        (button
          :onclick "hyprctl dispatch workspace ${i.id}"
          :class "${i.class}"
          "${replace(i.name, ':', '')}")))))
```

Currently that looks like this:

![20230713_19h57m53s_grim](https://github.com/elkowar/eww/assets/8584649/de190ca0-9a4b-432d-94fc-1b516400dd60)

What is bugging me is that all the buttons resize to match the size of the largest button. This isn't a huge deal since I'm using a monospace font, but when I added window titles to the indicators all the empty space started annoying me. In the two screenshots you can see that the button sizes are different if I have workspaces 1 2 3, or 1 2 10, where in the second instances all buttons grow to the width of 10, whereas I'd like them to all be sized to auto fit their content. Not a huge issue at the moment where I'm just showing monospace numbers, but I'd like to add window titles to it, and in that scenario it is annoying having all the dead space.

The second issue that I imagine is related is that I have another widget next my workspace indicators to launch a couple of rofi variants. The container for these buttons grows in a similiar fashion. This looks like this:

![20230713_19h58m28s_grim_c](https://github.com/elkowar/eww/assets/8584649/370e1bdf-3b64-4102-a04d-2d45cffc9f43)
![20230713_19h59m53s_grim](https://github.com/elkowar/eww/assets/8584649/3e675572-1aad-4bfe-94c5-ca14b73b3bc4)

The config of the relevant parts of my bar is roughly this:

```
(defwindow bar
  (barContainer))

(defwidget barContainer []
  (centerbox :orientation "h"
    (leftContainer)
    (centerContainer)
    (rightContainer)))

(defwidget leftContainer []
  (box :class "container" :width 1 :halign "start" 
     (workspaces0)
     (launchers)))

(defwidget workspaces0 []
  (eventbox :onscroll "hyprctl dispatch workspace `echo {} | sed 's/up/+/\' | sed 's/down/-/'`1"
    (box :class "windowManager" :halign "start" :spacing 10 :space-evenly: "false"
      (for i in workspace0
        (button
          :onclick "hyprctl dispatch workspace ${i.id}"
          :class "${i.class}"
          "${replace(i.name, ':', '')}")))))

(defwidget launchers[]
    (box :class "launchers" :space-evenly "false" :orientation "h" :spacing 10
     (applauncher)
     (syssearch)))
```

Would really apprecite some help, I assume I'm missing something obvious, but spent a few hours digging and wasn't able to solve it. Thanks.

**Comment by viandoxdev** [comment, marked as answer]
See #718, you have an extra colon after the `:space-evenly` in workspace0, and I'm not sure about the quoted booleans, you might want to remove the `"`s.

**Reply by sina--** [reply]
Thank you so much, the quotations were it. Figured there was something I just wasn't seeing - I appreciate it.
